test/e2e/suites/apivalidations/openstackmachine_test.go

```diff
@@ -50,7 +50,7 @@ var _ = Describe("OpenStackMachine API validations", func() {
 		Expect(k8sClient.Create(ctx, defaultMachine())).To(Succeed(), "OpenStackMachine creation should succeed")
 	})
 
-	It("should only allow the providerID to be set once", func() {
+	It("should only allow the providerID to be set once and identityRef to be set several times", func() {
```

Comment:
Please can you add a separate test for identityRef instead of adding it to this one? Unlike the e2e tests these tests are pretty fast to initialise, so it's ok to run lots of them. It's simpler to understand and maintain if they each do one thing only.